scripts/slurmsync.py

```diff
@@ -144,6 +145,18 @@ def find_node_status(nodename):
         if not state.base.startswith("DOWN"):
             return NodeStatus.terminated
     elif (state is None or "POWERED_DOWN" in state.flags) and inst.status == "RUNNING":
```

Comment:
We probably want to add both `PROVISIONING` and `STAGING` along `RUNNING` to this condition.
See https://cloud.google.com/compute/docs/instances/instance-life-cycle

Reply:
`state is None or "POWERED_DOWN" in state.flags` this seems to be too broad.

We could probably restrict it to `state.base == "DOWN" and "POWERED_DOWN" in state.flags`
See: https://slurm.schedmd.com/node_lifecycle.png

Reply:
1. I would push back to limiting state.base == "DOWN". I think if state were `State=IDLE+CLOUD+POWERED_DOWN` and the GCE VM was still live it would also be an orphan node (note base state of IDLE).

2. I have extensively tested the proposed solution. It solves the issue of killing wrongfully orphan nodes on startup without actually changing any state logic. I would prefer to submit this as is and then if we want to change the state logic do that independently.